P2: Add branding to Reader > Settings > Followed Sites

mindgraffiti opened the pull request:
Fixes #15473 

This PR fixes the bug found in #15473 and updates the default site icons for P2 sites.

### To test
1. Navigate to Reader tab > Settings (gear icon) > Followed sites.
2. Scroll for a bit.
3. Notice that site icons that contain transparencies are no longer tinged gray, but instead let the background color come through.

### Screenshots

| Bug | Fix in light mode | fix in dark mode |
| --- | --- | --- |
| ![Simulator Screen Shot - iPhone 11 Pro - 2020-12-07 at 16 15 38](https://user-images.githubusercontent.com/1062444/101922067-4e034300-3b93-11eb-895c-5df94375353a.png) | ![Simulator Screen Shot - iPhone 11 Pro - 2020-12-11 at 08 59 15](https://user-images.githubusercontent.com/1062444/101922094-55c2e780-3b93-11eb-8969-8f0dade7a332.png) | ![Simulator Screen Shot - iPhone 11 Pro - 2020-12-11 at 08 59 43](https://user-images.githubusercontent.com/1062444/101922111-58bdd800-3b93-11eb-92ee-006348786eae.png) |


PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

mindgraffiti committed:
Fix the gray tint color on site icons in Followed Sites list

mindgraffiti committed:
Merge branch 'develop' into feature/15264-p2-branding-mark-iii

mindgraffiti committed:
Merge branch 'develop' into feature/15264-p2-branding-mark-iii

mindgraffiti committed:
Check if the site is P2 type

mindgraffiti committed:
add release note

ScoutHarris commented:
Question - should the P2 icon have a border? It seems odd in this context where none of the others do.

<kbd><img width="53" alt="Screen Shot 2020-12-11 at 11 23 09 AM" src="https://user-images.githubusercontent.com/1816888/101940189-595e6a80-3ba3-11eb-8433-5ca197af6560.png"></kbd>

mindgraffiti committed:
Remove border, it's not consistent with any other site icon

mindgraffiti commented:
@ScoutHarris good point. In My Sites (site picker list) the border already existed, so I updated the color, but that's not the case for this view. Updated to remove the border ✅

ScoutHarris commented on WordPress/Classes/ViewRelated/Reader/ReaderFollowedSitesViewController.swift:
```diff
@@ -394,10 +394,16 @@ extension ReaderFollowedSitesViewController: WPTableViewHandlerDelegate {
+        var placeholderImage: UIImage = .siteIconPlaceholder
+        if site.isP2Type {
+            placeholderImage = UIImage.gridicon(.p2, size: CGSize(width: 40, height: 40))
+            cell.imageView?.layer.borderColor = UIColor.divider.cgColor
```
If it doesn't have a border, it doesn't need a border color, right?

ScoutHarris replied:
Also, if the border color can be removed, creating the `placeholderImage` can probably be reduced to something like:
`let placeholderImage = site.isP2Type ? UIImage.gridicon(.p2, size: CGSize(width: 40, height: 40)) : .siteIconPlaceholder`

mindgraffiti replied:
maybe I was saving it for later, who knows 😁 
deleted ✅

ScoutHarris reviewed (approved):
Just one more minor question, then :shipit: !

mindgraffiti committed:
Oops. Left the border color in there even though there is no border
